### ISSUE TITLE
Encoding sometimes hangs when using lavfi-colormatrix video filter.

### ISSUE THREAD

**Issue by ekisu** [mpv-player/mpv#6818]
### mpv version and platform

Windows 10.
```
mpv 0.29.0-353-g65b1c2d065 Copyright © 2000-2019 mpv/MPlayer/mplayer2 projects
 built on Sun Jul 14 20:16:49 +08 2019
ffmpeg library versions:
   libavutil       56.30.100
   libavcodec      58.53.101
   libavformat     58.28.102
   libswscale      5.4.101
   libavfilter     7.56.101
   libswresample   3.4.100
ffmpeg version: git-2019-07-13-f86f213a
```

### Reproduction steps

1. Attempt to do any encode using the `lavfi-colormatrix` video filter.

### Expected behavior

The encode should finish successfully every time.

### Actual behavior

Sometimes, it will just hang forever.

### Log file

[logenc.txt](https://github.com/mpv-player/mpv/files/3418302/logenc.txt)

### Sample files

I've used a script to test this issue, along with a blank video file ([link](https://files.catbox.moe/2v935e.zip)). Also, from what I've tested, this issue does not appear on older mpv versions, namely the following one:
```
mpv 0.29.0-329-g64cdc3694e Copyright © 2000-2019 mpv/MPlayer/mplayer2 projects
 built on Sun May 19 23:01:58 DST 2019
ffmpeg library versions:
   libavutil       56.28.100
   libavcodec      58.52.101
   libavformat     58.27.103
   libswscale      5.4.101
   libavfilter     7.53.101
   libswresample   3.4.100
ffmpeg version: N-45774-gfbdb3aa179
```

**Comment by onomatopellan** [comment]
There was a deadlock in winpthreads that should be fixed in latest build mpv-x86_64-20190810-git-639ee55.7z.

**Comment by ekisu** [comment]
Indeed, with this new build I can't reproduce the bug anymore. Thanks!
